crates/ar-cli/src/cli.rs

```diff
@@ -18,43 +19,100 @@ pub enum Endianess {
 
 #[derive(Debug, Subcommand)]
 pub enum Command {
+    /// Generate symmetric encryption keys
     Generate {
-        /// Directory where the generates keys are to be written to.
+        /// Directory where the generated keys are to be written to.
         ///
         /// Existing files will be overwritten!
         #[arg(long, default_value=get_default_dir())]
         dir: PathBuf,
 
         /// 32-byte hex seed.
         ///
-        /// Can contain `0x` prefix, which will be stripped.
+        /// if it contains `0x` prefix it will be stripped.
         /// Example: --seed 3fd54831f488a22b28398de0c567a3b064b937f54f81739ae9bd545967f3abab
-        #[arg(long, value_parser = ValueParser::new(parse_hex_as_seed))]
+        #[arg(long, value_parser = ValueParser::new(parse_32byte_array))]
         seed: [u8; 32],
 
         /// Should the output be produced in the Lower Endian (the default) or Big Endian order?
         #[clap(long, value_enum, default_value_t=Endianess::default())]
         endianess: Endianess,
     },
 
+    /// Read newAccount on-chain transactions and collect viewing keys
+    IndexEvents {
+        #[clap(flatten)]
+        common: Common,
+
+        #[clap(long, default_value = "10000")]
+        batch_size: usize,
+
+        #[clap(flatten)]
+        db: Db,
+    },
+
+    /// Read newAccount on-chain transactions and collect viewing keys
+    CollectKeys {
+        #[clap(flatten)]
+        common: Common,
+
+        #[clap(flatten)]
+        db: Db,
+
+        #[arg(long, default_value = "./private_key.bin")]
+        private_key_file: String,
+
+        /// is the key in the Lower Endian (the default) or Big Endian order?
+        #[clap(long, value_enum, default_value_t=Endianess::default())]
+        endianess: Endianess,
+    },
+
+    /// Matches everything we have in the db about the identity of the transactions
     Revoke {
-        #[arg(long)]
-        id: String,
+        #[clap(flatten)]
+        db: Db,
+    },
+
+    /// Given a tx-hash reveal all the txs with the same viewing_key
+    Reveal {
+        #[clap(flatten)]
+        db: Db,
+
+        #[arg(long, value_parser = ValueParser::new(parse_32byte_array))]
+        tx_hash: [u8; 32],
     },
 }
 
-fn parse_hex_as_seed(input: &str) -> Result<[u8; 32], &'static str> {
-    let mut decoded = [0u8; 32];
+#[derive(Debug, Args)]
+pub struct Db {
+    #[arg(long, default_value = "./ar_db.db3")]
+    pub path: PathBuf,
+}
+
+#[derive(Debug, Args)]
+pub struct Common {
```

Comment:
suggested name: `ChainConfig` or something around this, but feel free to ignore

Reply:
done